### PR TITLE
Add role bindings and SecurityContextConstraints as needed by kubedb and tekton

### LINKED ISSUE #80: Missing scc rule to create the PostgreSQL Pod
## Issue

When the `PostgreSQL Pod` will be created by the `statefulset` resource of the KubeDB operator, then the following error will be reported if the ServiceAccount of the POD don't has at least the `Privileged` SCC

```
Error creating: pods "pod..." is forbidden: unable to validate against any security context constraint: [capabilities.add: Invalid value: "IPC_LOCK": capability may not be added capabilities.add: Invalid value: "IPC_LOCK": capability may not be added]
```

The workaround is to include the SA used by the statefulset to the SCC

Example of SCC resource including the SA used by the KubeDB operator

```
apiVersion: v1
kind: SecurityContextConstraints
metadata:
  annotations:
    kubernetes.io/description: 'privileged allows access to all privileged and host
      features and the ability to run as any user, any group, any fsGroup, and with
      any SELinux context.  WARNING: this is the most relaxed SCC and should be used
      only for cluster administration. Grant with caution.'
  name: privileged
readOnlyRootFilesystem: false
runAsUser:
  type: RunAsAny
seLinuxContext:
  type: RunAsAny
seccompProfiles:
- '*'
supplementalGroups:
  type: RunAsAny
users:
- system:admin
- system:serviceaccount:openshift-infra:build-controller
- system:serviceaccount:openshift-node:sync
- system:serviceaccount:openshift-sdn:sdn
- system:serviceaccount:management-infra:management-admin
- system:serviceaccount:management-infra:inspector-admin
- system:serviceaccount:istio-system:jaeger
- system:serviceaccount:gandrian:default
- system:serviceaccount:istio-system:istio-pilot-service-account
- system:serviceaccount:gandrian:sa-greeting
- system:serviceaccount:openshift-logging:aggregated-logging-fluentd
- system:serviceaccount:cmoullia:default
- system:serviceaccount:cmoullia:build-bot
- system:serviceaccount:demo:build-bot
- system:serviceaccount:demo:my-postgres
- system:serviceaccount:demo:postgres-db
- system:serviceaccount:test1:postgres-db
volumes:
- '*'
allowHostDirVolumePlugin: true
allowHostIPC: true
allowHostNetwork: true
allowHostPID: true
allowHostPorts: true
allowPrivilegeEscalation: true
allowPrivilegedContainer: true
allowedCapabilities:
- '*'
defaultAddCapabilities: null
fsGroup:
  type: RunAsAny
groups:
- system:cluster-admins
- system:nodes
- system:masters
```

### PR DESCRIPTION
**Pull request by metacosm**
Should fix #80.
Also took the opportunity to rationalize how names are generated and used so that we don't use the same names by coincidence but rather make the intent explicit.

### CODE REVIEW

File: pkg/controller/component/build_deployment.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func (res deployment) installBuild() (runtime.Object, error) {
 	// and we will enrich the deployment resource of the runtime container
 	// create a "dev" version of the component to be able to check if the dev deployment exists
 	devDeployment := &appsv1.Deployment{}
-	_, err = res.reconciler.Helper().Fetch(defaultNamer(c), c.Namespace, devDeployment)
+	_, err = res.reconciler.Helper().Fetch(DeploymentNameFor(c, v1alpha2.DevDeploymentMode), c.Namespace, devDeployment)
```
Comment (cmoulliard) [inline pkg/controller/component/build_deployment.go]: So here we will fetch the Deployment having the name `component_name` or `component_name-build` - Correct ?

**Comment by metacosm** [comment]
No, this change only fetches the dev deployment as was the case before. This doesn’t address #102 at all.

**Comment by cmoulliard** [comment]
LGTM. Can be merged

File: pkg/controller/k8srole.go [hunk 2]
```diff
@@ -9,21 +9,22 @@ import (
 
 type k8srole struct {
 	*DependentResourceHelper
+	reconciler *BaseGenericReconciler
```
Comment (metacosm) [inline pkg/controller/k8srole.go]: Why are you injecting the reconciler when it doesn't appear to be used anywhere in the code?

File: pkg/controller/k8srole.go [hunk 3]
```diff
@@ -9,21 +9,22 @@ import (
...
 }
 
 func (res k8srole) Update(toUpdate runtime.Object) (bool, error) {
 	return false, nil
 }
 
 func (res k8srole) NewInstanceWith(owner v1alpha2.Resource) DependentResource {
-	return newOwnedk8srole(owner)
+	return newOwnedk8sRole(res.reconciler, owner)
```
Comment (metacosm) [inline pkg/controller/k8srole.go]: `newOwnedk8sRole` => `newOwnedK8sRole` would be better, no?

Comment (metacosm) [reply pkg/controller/k8srole.go]: Same for other function names, actually…

Comment (cmoulliard) [reply pkg/controller/k8srole.go]: I agree with you ;-)

File: pkg/controller/k8srole.go (hunk 2 above)
Comment (cmoulliard) [reply pkg/controller/k8srole.go]: I simply adapted the code according to last code committed on issue-80 but didn't has the chance to check if it was used, needed, ...

Comment (cmoulliard) [reply pkg/controller/k8srole.go]: Will remove it then

**Pull request merged**